deepmd/pt/model/descriptor/sezm_nn/triton/so2_rotation.py

```
@@ -0,0 +1,1715 @@
+        o_sr,
+        o_sc,
+        LMAX: tl.constexpr,
+        BLOCK_C: tl.constexpr,
+    ):
+        """``out[e,(l,m),c] = sum_{j} W[e, l^2+l+m, l^2+j] * x[src[e], l^2+j, c]``."""
+        edge = tl.program_id(0).to(tl.int64)
+        chan = tl.arange(0, BLOCK_C)
+        cmask = chan < channels
+        src_idx = tl.load(src_ptr + edge).to(tl.int64)
+
+        for l in tl.static_range(0, LMAX + 1):
```

Ruff (0.15.15)

[error] Ambiguous variable name: `l`

(E741)